internal/image/image.go

```diff
@@ -30,61 +31,112 @@ type ScanResults struct {
 }
 
 type Image struct {
-	flattenedLayers []fileMap
-	innerImage      *v1.Image
-	extractDir      string
+	// Final layer is the last element in the slice
+	layers         []imgLayer
+	innerImage     *v1.Image
+	extractDir     string
+	baseImageIndex int
+	configFile     *v1.ConfigFile
+	layerIDToIndex map[string]int
 }
 
-func (img *Image) LastLayer() fileMap {
-	return img.flattenedLayers[len(img.flattenedLayers)-1]
+func (img *Image) LayerIDToIdx(id string) int {
```

Comment:
nit: instead of `Idx`, let's just use the full `Index` for this function name. e.g. https://pkg.go.dev/strings#Index

Reply:
Function has been removed entirely as it's not needed anymore